### PR TITLE
Update python 3.9 to 3.11

### PR DESCRIPTION
**Pull request by AviGawande**
This PR closes/references issue #117 . It does so by …

## Checklist

- [x] I have added tests to cover my changes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,25 +20,24 @@ RUN apt-get update && \
...
 COPY pyproject.toml /pretalx
 COPY src /pretalx/src
 COPY deployment/docker/pretalx.bash /usr/local/bin/pretalx
 COPY deployment/docker/supervisord.conf /etc/supervisord.conf
 
-RUN pip3 install -U pip setuptools wheel typing && \
-    pip3 install -e /pretalx/[mysql,postgres,redis] && \
-    pip3 install pylibmc && \
-    pip3 install gunicorn
+RUN pip install -U pip setuptools wheel typing && \
```
Comment (hongquan) [inline Dockerfile:28]: These changes (removing suffix `3`) are unnecessary and even less safe.

**Review by hongquan — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (AviGawande) [reply Dockerfile]: yes @hongquan fixed the changes

**Review by hongquan — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -26,15 +26,15 @@ jobs:
...
         database: [sqlite, postgres, mysql]
         exclude:
           - database: mysql
-            python-version: 3.9
+            python-version: 3.11
           - database: mysql
             python-version: 3.11
           - database: sqlite
-            python-version: 3.9
+            python-version: 3.11
```
Comment (norbusan) [inline .github/workflows/tests.yml:37]: Why do you exclude the version 3.11 / sqlite tests? 
The orginal coede had tests matrix 3.9 / 3.10 / 3..11  and only 3.9/sqlite was excluded.

I think all the exclude steps should be removed!!!

**Review by norbusan — CHANGES_REQUESTED** [review]

**Comment by norbusan** [comment]
Some changes to the .rst files that mention Python 3.9 might also be a good idea.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (AviGawande) [reply .github/workflows/tests.yml]: fixed the changes

**Comment by AviGawande** [comment]
> Some changes to the .rst files that mention Python 3.9 might also be a good idea.

yes i updated it as well

**Review by hongquan — APPROVED** [review]

**Pull request merged**